app/v2/notifications/rest_push.py

```diff
@@ -1,59 +1,99 @@
     the Flask "request" instance.
     """
-
-    if not is_feature_enabled(FeatureFlag.PUSH_NOTIFICATIONS_ENABLED):
-        raise NotImplementedError()
-
     if not authenticated_service.has_permissions(PUSH_TYPE):
         raise BadRequestError(
-            message='Service is not allowed to send {}'.format(get_public_notify_type_text(PUSH_TYPE, plural=True))
+            message='Service is not allowed to send {}'.format(
+                get_public_notify_type_text(PUSH_TYPE, plural=True),
+            ),
+            status_code=403,
         )
 
-    req_json = validate(request.get_json(), schema)
+    payload = validate_push_payload(schema)
 
-    if 'mobile_app' in req_json:
-        app_instance = mobile_app_registry.get_app(MobileAppType[req_json['mobile_app']])
+    try:
+        # Choosing to use the email queue for push to limit the number of empty queues
+        deliver_push.delay(payload)
+    except (CeleryError, OperationalError):
+        current_app.logger.exception('Failed to enqueue deliver_push request')
+        response = jsonify(result='error', message='VA Notify service impaired, please try again'), 502
     else:
-        app_instance = mobile_app_registry.get_app(DEAFULT_MOBILE_APP_TYPE)
+        response = jsonify(result='success'), 201
+    # Flask turns the tuple into a json and status_code
+    return response
+
+
+def validate_push_payload(schema: dict[str, str]) -> V2PushPayload:
+    """Validate an incoming push request.
+
+    Args:
+        schema (dict[str, str]): The incoming request
 
+    Raises:
+        BadRequestError: Failed validation
+
+    Returns:
+        dict[str, str]: Validated request dictionary
+    """
     try:
-        vetext_client.send_push_notification(
-            app_instance.sid,
-            req_json['template_id'],
-            req_json['topic_sid'] if is_broadcast else req_json['recipient_identifier']['id_value'],
-            is_broadcast,
-            req_json.get('personalisation'),
-        )
-    except VETextBadRequestException as e:
-        raise BadRequestError(message=e.message, status_code=400)
-    except (VETextNonRetryableException, VETextRetryableException):
-        return jsonify(result='error', message='Invalid response from downstream service'), 502
-    else:
-        return jsonify(result='success'), 201
+        req_json: dict[str, str] = validate(request.get_json(), schema)
+
+        # Validate the application they sent us is valid or use the default
+        if 'mobile_app' in req_json:
+            app_sid = mobile_app_registry.get_app(MobileAppType[req_json['mobile_app']]).sid
+        else:
+            app_sid = mobile_app_registry.get_app(DEAFULT_MOBILE_APP_TYPE).sid
+    except (KeyError, TypeError) as e:
+        current_app.logger.warning('Push request failed validation due to mobile app setup: %s', e)
+        raise BadRequestError(message=str(e), status_code=400)
```

Comment:
Is this still relevant?  You deleted the call to send_push_notification, which might have raised KeyError via the calling parameters.

Reply:
Yes - `validate` method raises a ValidationError. We would like to include the logs below if a ValidationError is thrown.

Reply:
Remove TypeError - Keeping KeyError which is raised in `mobile_app_registry.get_app`

Reply:
I decided not to remove the TypeError due to issues with test `tests/app/integration/test_push_notification.py::test_mobile_app_push_notification_failed_validation`. I tried to update this test and remove testing for `TypeError`. While updating the test, `TypeError` continued to be thrown - appears a `TypeError` is thrown if `mobile_app` is None in the request_body.